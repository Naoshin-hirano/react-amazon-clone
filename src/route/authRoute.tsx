import { Route } from "@tanstack/react-location";
import React from "react";
import { LoginPage } from "ui/page/login";
import { RegisterPage } from "ui/page/register";

export const authRoute: Route[] = [
    {
        path: "/login",
        element: <LoginPage />,
    },
    {
        path: "/register",
        element: <RegisterPage />,
    },
];
