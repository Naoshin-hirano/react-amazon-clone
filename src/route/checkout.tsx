import { Route } from "@tanstack/react-location";
import React from "react";
// import { SocondSamplePage } from "ui/page/second";
import { CheckoutPage } from "ui/page/checkout";

export const checkoutRoute: Route[] = [
    {
        path: "/checkout",
        element: <CheckoutPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
