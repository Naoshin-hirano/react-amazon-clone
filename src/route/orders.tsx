import { Route } from "@tanstack/react-location";
import React from "react";
import { OrdersPage } from "ui/page/orders";

export const ordersRoute: Route[] = [
    {
        path: "/orders",
        element: <OrdersPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
