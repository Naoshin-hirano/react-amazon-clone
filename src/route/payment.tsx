import { Route } from "@tanstack/react-location";
import React from "react";
import { PaymentPage } from "ui/page/payment";

export const paymentRoute: Route[] = [
    {
        path: "/payment",
        element: <PaymentPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
