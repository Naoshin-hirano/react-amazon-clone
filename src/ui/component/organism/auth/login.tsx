import React from "react";
import "../../../../design/component/atoms/auth/Login.css";
import { Link, MemoryRouter } from "react-router-dom";

export const Login: React.FC<any> = (props) => {
    const { firstTitle } = props;
    console.log(firstTitle);

    return (
        <div className="login">
            <MemoryRouter>
                <Link to="/">
                    <img
                        className="login__logo"
                        src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
                    />
                </Link>
            </MemoryRouter>

            <div className="login__container">
                <h1>Sign In</h1>
                <form>
                    <h5>Email</h5>
                    <input type="text" />

                    <h5>Password</h5>
                    <input type="password" />

                    <button className="login_signInButton">Sign In</button>
                </form>

                <p>
                    By signing in you agree to Amazon FAKE CLONE's Conditions of
                    Use & Sale. Please see our Privacy Notice, or Cookies Notice
                    and our Interest-Based Ads Notice.
                </p>

                <button className="login__registerButton">
                    Create your Amazon Account
                </button>
            </div>
        </div>
    );
};
