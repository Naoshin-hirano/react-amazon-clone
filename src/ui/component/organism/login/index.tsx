import React from "react";
import "../../../../design/component/atoms/auth/login.css";
import { Link } from "@tanstack/react-location";

export const Login: React.FC<any> = (props) => {
    const { firstTitle } = props;
    console.log(firstTitle);

    return (
        <div className="login">
            <Link to="/">
                <img
                    alt="Amazon_logo"
                    className="login__logo"
                    src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
                />
            </Link>
            <div className="login__container">
                <h1>ログイン</h1>
                <form>
                    <h5>メール</h5>
                    <input type="text" />

                    <h5>パスワード</h5>
                    <input type="password" />

                    <button className="login_signInButton">ログイン</button>
                </form>

                <p>
                    サインインすることにより、Amazon FAKE CLONE
                    の条件に同意したことになります
                    使用と販売。当社のプライバシーに関する通知または Cookie
                    に関する通知をご覧ください。 およびインタレスト
                    ベース広告の通知。
                </p>

                <button className="login__registerButton">
                    Amazon Accountを作成する
                </button>
            </div>
        </div>
    );
};
