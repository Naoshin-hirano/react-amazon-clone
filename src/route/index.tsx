import { ReactLocation, Route } from "@tanstack/react-location";
import { history } from "./history";
import { authRoute } from "route/authRoute";
import React from "react";
import { TopPage } from "ui/page/top";

const indexRoute: Route[] = [
    {
        path: "/",
        element: <TopPage />,
    },
    {
        path: "/top",
        element: <TopPage />,
    },
];

export const routes: Route[] = [authRoute].reduce(
    (prevRoute, currRoute) => prevRoute.concat(currRoute),
    indexRoute
);

export const location = new ReactLocation({ history: history });
