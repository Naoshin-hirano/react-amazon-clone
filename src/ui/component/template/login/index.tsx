import React from "react";
import { Login } from "ui/component/organism/auth/login";

export const Template: React.FC<any> = (props) => {
    return (
        <div>
            <Login {...props} />
        </div>
    );
};
