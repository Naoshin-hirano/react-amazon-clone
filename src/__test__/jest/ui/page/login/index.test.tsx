import { render } from "@testing-library/react";
import React from "react";
import { Router } from "@tanstack/react-location";
import { location, routes } from "../../../../../route";
import { LoginPage } from "ui/page/login";

describe("LoginPage", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("Login画面 snapshot確認", async () => {
        const { asFragment } = render(
            <Router routes={routes} location={location}>
                <LoginPage />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot("LoginPage");
    });
});
