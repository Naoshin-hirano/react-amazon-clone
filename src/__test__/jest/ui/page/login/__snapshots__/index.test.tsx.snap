// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Login画面 snapshot確認: LoginPage 1`] = `
<DocumentFragment>
  <div>
    <div
      class="login"
    >
      <a
        class="active"
        href="/"
      >
        <img
          alt="Amazon_logo"
          class="login__logo"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
        />
      </a>
      <div
        class="login__container"
      >
        <h1>
          ログイン
        </h1>
        <form>
          <h5>
            メール
          </h5>
          <input
            type="text"
          />
          <h5>
            パスワード
          </h5>
          <input
            type="password"
          />
          <button
            class="login_signInButton"
          >
            ログイン
          </button>
        </form>
        <p>
          サインインすることにより、Amazon FAKE CLONE の条件に同意したことになります 使用と販売。当社のプライバシーに関する通知または Cookie に関する通知をご覧ください。 およびインタレスト ベース広告の通知。
        </p>
        <button
          class="login__registerButton"
        >
          Amazon Accountを作成する
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
